controller/bluegreen.go

```diff
@@ -248,6 +248,20 @@ func (c *Controller) scaleBlueGreen(rollout *v1alpha1.Rollout, newRS *appsv1.Rep
 		}
 	}
 
+	previewRS, _ := replicasetutil.GetReplicaSetByTemplateHash(allRS, rollout.Status.BlueGreen.PreviewSelector)
+	if previewRS != nil {
+		var previewReplicas int32
+		if rollout.Spec.Strategy.BlueGreenStrategy.PreviewReplicaCount != nil {
```

Comment:
We need to take into account when the `status.blueGreenStatus.scaleUpPreviewCheckPoint` field is set to true. That field is used to indicates that the rollout has been unpaused and it should scale up the preview ReplicaSet to the full Replica count before switching the active service to it.

Reply:
Thanks I thought about that but didn't know if there was a good way to know.